### ISSUE TITLE
Gerrit Authentication Cookie

### ISSUE THREAD

**Issue by venth** [google/concourse-resources#2]
Hi,

would you like to provide an example of a cookie that can be used for authentication purposes?

**Comment by lann** [comment]
It isn't well documented, but if your Gerrit instance is configured for it you can get instructions by logging in and going to Settings -> HTTP Credentials -> Obtain password. You can see an example of this on the Gerrit project's own instance: https://gerrit-review.googlesource.com/new-password

**Comment by lann** [comment]
I would also accept a patch (or will get around to it eventually™) adding other auth support. The underlying gerrit library already supports this: https://godoc.org/golang.org/x/build/gerrit#Auth

**Comment by venth** [comment]
It seems that I need to think about patch to provide authentication capabilities :( Thanks for the explanation.

**Comment by venth** [comment]
Do you use glide or any other vendoring tool?

**Comment by lann** [comment]
I have been using [golang/dep](https://github.com/golang/dep).

**Comment by venth** [comment]
What do you think, is it better to replace: 
* `cookies` with
  ```
  auth:
     cookies
  ```
and extend authentication methods by:
  ```
  auth:
    user:
    password:
  ```
* or just extend model by optional:
```
   user:
   password:
```
* any other suggestions?

**Comment by lann** [comment]
Looking at the [existing Concourse resources](https://github.com/concourse/git-resource#source-configuration), I think top level `username` and `password` fields would be most consistent.

**Comment by venth** [comment]
OK, I've started to work on that

**Comment by venth** [comment]
Introduced in [pull request](https://github.com/google/concourse-resources/pull/5)

**Comment by lann** [comment]
Issue addressed by #6 (username/password auth).
